src/python/WMCore/WMSpec/Steps/Executors/CMSSW.py

```diff
@@ -106,12 +111,20 @@ def execute(self, emulator=None):
         self.failedPreviousStep = overrides.get('previousCmsRunFailure', False)
 
         if self.failedPreviousStep:
+            # use logic of _setStatus which assigns
```

Comment:
I am inclined to say that all these changes under this method are not really necessary. And that is because we now have an `if` statement in the `_setStatus` method.

Reply:
I'm not sure about it since the current `if self.failedPreviousStep` block through exception which abort the job (my understanding). Do you want to remove entire block? Please clarify your comment further.

Reply:
No, my suggestion is only to remove code that you added in this PR. Keeping the previous logic in place.

Reply:
I restored original logic but still  I removed from it `self._setStatus(99108, msg)` since it is wrong code and it should not be set over here. Please review it again and let me know.